### PR TITLE
Add cstruct conversion

### LINKED ISSUE #5: Add to/from Cstruct
See conversation: https://github.com/mirleft/ocaml-nocrypto/issues/50

Are you all open to this being added? I was thinking something like:

val of_cstruct : Cstruct.t -> t
val to_cstruct : t -> Cstruct.t

I also like @pqwy comments about ignore chars, so the above could instead be:

val of_cstruct : ?ignore:char list -> Cstruct.t -> t

Am happy to submit a PR. Thanks.

### PR DESCRIPTION
**Pull request by trevorsummerssmith**
Meant to close #5 

Printing functions are added in another pull request.

### CODE REVIEW

**Comment by trevorsummerssmith** [comment]
Any thoughts on this? Thanks.

File: lib/hex.ml [hunk 1]
```diff
@@ -34,37 +35,57 @@ let to_char x y =
...
+let of_cstruct ?(ignore=[]) cs =
+  let open Cstruct in
+  of_helper
+    ~ignore:ignore
+    (fun i -> Bigarray.Array1.get cs.buffer (cs.off+i))
+    cs.len
+
+let to_cstruct hex =
+  to_helper
+    ~empty_return:(Cstruct.of_string "")
```
Comment (samoht) [inline lib/hex.ml]: You are allocating an empty Cstruct here for every call to `to_cstruct`. Would be better to allocate it once globally I think (as it is empty nobody can modify it anyway).

**Comment by samoht** [comment]
Sorry I completely miss your 2 PRs. I made a few comments but they look good to me. Can you fix Travis CI by adding `cstruct` as a dependency in the `opam` file as well? Thanks!

**Comment by trevorsummerssmith** [comment]
Updated the cstruct to only allocate once. Added the opam dependency.

**Comment by samoht** [comment]
Thanks!

**Pull request merged**